src/Microsoft.OData.Client/BatchSaveResult.cs

```diff
@@ -288,7 +301,14 @@ private ODataRequestMessageWrapper CreateBatchRequest()
             Uri requestUri = UriUtil.CreateUri(this.RequestInfo.BaseUriResolver.GetBaseUriWithSlash(), UriUtil.CreateUri("$batch", UriKind.Relative));
             HeaderCollection headers = new HeaderCollection();
             headers.SetRequestVersion(Util.ODataVersion4, this.RequestInfo.MaxProtocolVersionAsVersion);
-            headers.SetHeader(XmlConstants.HttpContentType, CreateMultiPartMimeContentType());
+            if(useJsonBatch)
+            {
+                headers.SetHeader(XmlConstants.HttpContentType, CreateApplicationJsonContentType());
+            }
+            else
+            {
+                headers.SetHeader(XmlConstants.HttpContentType, CreateMultiPartMimeContentType());
+            }

```

Comment:
What happens when the client sends `IEEE754Compatible` `Json `request?

Reply:
@paulodero In the above code, we are setting headers for the batch `top level request`.
`IEEE754Compatible Json` request headers are in the individual requests

Reply:
I get it.